scipy/ndimage/_measurements.py

```diff
@@ -630,6 +630,10 @@ def single_group(vals):
     if labels is None:
         return single_group(input)
 
+    # manually cast to numpy
+    # since libaries (e.g. dask) that implement __array_function__
+    # will not return a numpy array from broadcast_arrays
```

Comment:
it's less complicated than this; you converted input to numpy (thus disregarding array API completely ) therefore labels needs to match.

Reply:
That's true, but I thought it'd be good to document this behavior.
(since normally you can rely on numpy automatically using ``__array__`` to convert the inputs into a numpy array)